test/e2e/common.go

```diff
@@ -78,6 +78,12 @@ func dumpSpecResourcesAndCleanup(ctx context.Context, specName string, clusterPr
 		LogPath:   filepath.Join(artifactFolder, "clusters", clusterProxy.GetName(), "resources"),
 	})
 
+	// Dump kube-system pods in the workload cluster before deleting it.
```

Comment:
Might need a flag in `dumpSpecResourcesAndCleanup` to enable / disable this. The runtime-sdk job does the delete call in-line so it's failing because the cluster doesn't exist at this point.

Reply:
I'll try to auto-discover by checking if the Cluster still exists